htp/htp_response_generic.c

```diff
@@ -59,7 +59,7 @@ htp_status_t htp_parse_response_line_generic(htp_connp_t *connp) {
     tx->response_message = NULL;
 
     // Ignore whitespace at the beginning of the line.
-    while ((pos < len) && (htp_is_space(data[pos]))) pos++;
+    while ((pos < len) && (htp_is_space(data[pos]) || data[pos] == 0)) pos++;
```

Comment:
can you explain why this is needed and what the consequences are? Suricata has a 'http_raw_headers' keyword that expects raw data, so including unprintable chars I think

Reply:
This is in fact not needed here, I remove it

A few lines below, the null characters check is needed for meaningful headers such as content-length.
When libhtp does `cl = htp_table_get_c(connp->out_tx->response_headers, "content-length")`
it does not work if it is `"content\x00-length"` instead of `"content-length"` but browsers do recognize them

So, is this ok ?

Reply:
Ok can you make sure the commit message clearly explains this? Also that it is about the header names, not values.

Reply:
In fact, it is for both header names and header values.
As I remember, we have to do it for values for the case
`Transfer-encoding: chun\x00ked`

So, what is the expected behavior ?
Normalize names, but not values, and use a new `strcmp_with_zeroes` function for checking if transfer-encoding value is chunked ?

Reply:
From Suricata's perspective, I would expect the following:

1. http_header keyword works on normalized keys and values, so nulls should be stripped
2. http_raw_header keyword should have the nulls in both keys and values
3. the nulls should lead to 'event' we can match on

Reply:
I actually think that there should be 2 events: one for nulls in the keys and the other for nulls in the values. The former seems more serious to me.

Reply:
Ok, the last version removed the warning events. I will add them again when I get back to this